react/components/molecules/select/select.js

```diff
@@ -0,0 +1,146 @@
+import React, { PureComponent } from "react";
+import { StyleSheet, ViewPropTypes, View } from "react-native";
+import PropTypes from "prop-types";
+import RNPickerSelect from "react-native-picker-select";
+import { mix } from "yonius";
+
+import { IdentifiableMixin, baseStyles, equal } from "../../../util";
+
+import { Icon } from "../../atoms";
+
+export class Select extends mix(PureComponent).with(IdentifiableMixin) {
+    static get propTypes() {
+        return {
+            options: PropTypes.array,
+            value: PropTypes.any,
+            placeholder: PropTypes.string,
+            disabled: PropTypes.bool,
+            width: PropTypes.number,
+            onUpdateValue: PropTypes.func,
+            style: ViewPropTypes.style
+        };
+    }
+
+    static get defaultProps() {
+        return {
+            options: [],
+            value: undefined,
+            placeholder: undefined,
+            disabled: false,
+            width: undefined,
+            keyTimeout: 500,
+            onUpdateValue: () => {},
+            style: {}
+        };
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            valueData: props.value
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        if (!equal(prevProps.value, this.props.value)) {
+            this.setState({
+                valueData: this.props.value
+            });
+        }
+    }
+
+    onValueChange = value => {
+        this.setState(
+            {
+                valueData: value
+            },
+            () => this.props.onUpdateValue(value)
+        );
+    };
+
+    _icon = () => {

```

Comment:
Aren't we calling this stuff `renderIcon` and keeping it next to `render`? Not sure, just checking.

Reply:
Both in the components-vue and react-native, we are using `icon` and `_icon` respectively